common/network_manager.go

```diff
@@ -742,6 +742,22 @@ func (m *NetworkManager) SetOfflineMode(offline bool) {
 	}
 }
 
+// ThrottleNetwork changes the network attributes in chrome to simulate slower
+// networks e.g. a slow 3G connection.
+func (m *NetworkManager) ThrottleNetwork(networkProfile NetworkProfile) error {
+	action := network.EmulateNetworkConditions(
+		m.offline,
```

Comment:
If I understand this correctly, we are using here the `NetworkManager.offline` attribute, which is set when calling `BrowserContext.SetOfflineMode`. This implies that, if there is a previous call to `SetOfflineMode`, a call to `ThrottleNetwork` won't apply, as the page is being emulated as being "offline".  Is that correct?

I wonder if we should automatically overwrite the `offline` attribute to "false" when calling `ThrottleNetwork`, as the user is expressing an intention on applying this emulation to this page. Instead of requiring our users to call `BrowserContext.SetOffline(false)`, which will actually apply to all `BrowserContext` pages, not just this one.

Reply:
> This implies that, if there is a previous call to SetOfflineMode, a call to ThrottleNetwork won't apply, as the page is being emulated as being "offline". Is that correct?

Yes, that's correct.

> I wonder if we should automatically overwrite the offline attribute to "false" when calling ThrottleNetwork, as the user is expressing an intention on applying this emulation to this page. Instead of requiring our users to call BrowserContext.SetOffline(false), which will actually apply to all BrowserContext pages, not just this one.

That could work. My hesitation towards this idea is that the API is quite explicit at the moment to enable/disable offline mode (as well as the other APIs we support when it comes to enabling/disabling features such as `grantPermissions`). If we allow the offline mode to be automatically disabled on `ThrottleNetwork`, then this breaks the explicit nature of our APIs. I feel that being explicit (at least in the short term) is the best way forward until we see how it is being used.

Reply:
I see your point. Still, I'm not sure this is the better approach, maybe documenting the overwrite when calling these page level methods would make sense? Otherwise I feel we should inform the user somehow that the network throttling call will not apply, if `offline` is `true` when calling `ThrottleNetwork`, but I guess for that the only option is to return an error.

Reply:
> Otherwise I feel we should inform the user somehow that the network throttling call will not apply, if `offline` is `true` when calling `ThrottleNetwork`, but I guess for that the only option is to return an error.

I'm happy with this idea since it makes things more transparent and quickly highlights an issue preventing the `ThrottleNetwork` from working.

Reply:
Ok, to be honest I don't quite like the idea of returning an error from `ThrottleNetwork` call anyway, so let's follow your proposal. Still, I feel this might get some confusion, althought I don't think `setOfflineMode` is used frequently, so let's see what is user's feedback. 
Will we then need to store the current page network profile?

Reply:
> Will we then need to store the current page network profile?

Yes, correct, i'll do that next.

Reply:
Done in 24b9b97678dd30366b2b4b03e56cd1d71a8c4ef8. I'll refactor this so that `SetOfflineMode` will work with `ThrottleNetwork` in another PR. This change now works with the following example:

<details>
  <summary>test script</summary>

```js
import { browser } from 'k6/x/browser';

export const options = {
  scenarios: {
    throttled: {
      executor: 'shared-iterations',
      options: {
        browser: {
            type: 'chromium',
        },
      },
      exec: 'throttled',
    },
  },
}

export async function throttled() {
  const context = browser.newContext();
  const page = context.newPage();

  try {
    context.setOffline(true);

    page.throttleNetwork({
      latency: 1000,
      downloadThroughput: 50,
      uploadThroughput: 50,
    });

    context.setOffline(false); // Throttled network profile is kept to what was set in the line above.

    await page.goto('https://test.k6.io/', { waitUntil: 'networkidle' });
  } finally {
    page.close();
  }
}
```

</details>